module/Core/test/RedirectRule/Entity/RedirectConditionTest.php

```diff
@@ -153,4 +154,19 @@ public function canBeCreatedFromImport(string $type, RedirectConditionType|null
         );
         self::assertEquals($expectedType, $condition?->type);
     }
+
+    #[Test, DataProvider('provideVisitsWithBeforeDateCondition')]
+    public function matchesBeforeDate(string $date, bool $expectedResult): void
+    {
+        $request = ServerRequestFactory::fromGlobals();
+        $result = RedirectCondition::forBeforeDate($date)->matchesRequest($request);
+
+        self::assertEquals($expectedResult, $result);
+    }
+
+    public static function provideVisitsWithBeforeDateCondition(): iterable
+    {
+        yield 'date later than current' => [Chronos::now()->addHours(1)->toIso8601String(), true];
+        yield 'date earlier than current' => [Chronos::now()->subHours(1)->toIso8601String(), false];
```

Comment:
As per my previous comment, you can probably pass a raw Chronos object here.

```suggestion
        yield 'date later than current' => [Chronos::now()->addHours(1), true];
        yield 'date earlier than current' => [Chronos::now()->subHours(1), false];
```